aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +692,250 @@ public void onError(Exception error) {
         );
     }
 
+    @Override
+    public void fetchUserAttributes(
+            @NonNull Consumer<List<AuthUserAttribute>> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        awsMobileClient.getUserAttributes(new Callback<Map<String, String>>() {
+            @Override
+            public void onResult(Map<String, String> result) {
+                List<AuthUserAttribute> userAttributes = new ArrayList<>();
+                for (Map.Entry<String, String> entry : result.entrySet()) {
+                    userAttributes.add(new AuthUserAttribute(
+                            new AuthUserAttributeKey(entry.getKey()), entry.getValue()));
+                }
+                onSuccess.accept(userAttributes);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                onError.accept(new AuthException(
+                        "Failed to fetch user attributes since user is not authenticated",
+                        error,
+                        "See attached exception for details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void updateUserAttribute(
+            @NonNull AuthUserAttribute attribute,
+            @NonNull Consumer<AuthUpdateAttributeResult> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+
+        awsMobileClient.updateUserAttributes(
+                Collections.singletonMap(attribute.getKey().getKeyString(), attribute.getValue()),
+                new Callback<List<UserCodeDeliveryDetails>>() {
+                    @Override
+                    public void onResult(List<UserCodeDeliveryDetails> result) {
+                        if (!isValidUserAttribute(attribute.getKey().getKeyString())) {
+                            onError.accept(new AuthException(
+                                    "Invalid attribute name",
+                                    "See attached exception for more details"
+                            ));
+                        }
+
+                        if (!isUserAttributeVerifiedByCode(attribute.getKey().getKeyString())) {
```

Comment:
This isn't the best way to handle this because what if Cognito adds verification methods for other attributes in the future? What we know is that the onResult method only gets called if all attributes (in this case just the one) are updated successfully. The list type result object contains an entry for any attribute(s) which now need to be confirmed with a sent code and contains no entry for the others. So the logic here for updating one attribute should be as simple as:

onResult:
   if(result.size() == 0) return result with DONE status
   else return result with CONFIRM_ATTRIBUTE_WITH_CODE status

onError:
   return error